src/hotspot/share/memory/metaspace/blockTree.cpp

```diff
@@ -92,25 +109,28 @@ void BlockTree::verify() const {
 
       // Assume a (ridiculously large) edge limit to catch cases
       //  of badly degenerated or circular trees.
-      assrt0(info.depth < 10000);
+      tree_assert(info.depth < 10000, "too deep (%u)", info.depth);
       counter.add(n->_word_size);
 
       // Verify node.
+      tree_assert_invalid_node(n->_canary == Node::_canary_value, n);
+
       if (n == _root) {
-        assrt0(n->_parent == NULL);
+        tree_assert_invalid_node(n->_parent == NULL, n);
       } else {
-        assrt0(n->_parent != NULL);
+        tree_assert_invalid_node(n->_parent != NULL, n);
       }
 
       // check size and ordering
-      assrt(n->_word_size >= MinWordSize, "bad node size " SIZE_FORMAT, n->_word_size);
-      assrt0(n->_word_size > info.lim1);
-      assrt0(n->_word_size < info.lim2);
+      tree_assert_invalid_node(n->_word_size >= MinWordSize &&
+                               n->_word_size <= chunklevel::MAX_CHUNK_WORD_SIZE, n);
```

Comment:
I prefer one assert per line (that is, not using `&&` in asserts). In this case it does not matter as much, as we can deduce which sub-expression failed from the tree printout,  but I think it is a better style, because in _general_ you do not want such an assert to fail.